examples/java/spring-boot/src/test/java/uk/gov/api/springboot/infrastructure/fetcher/v1alpha/V1AlphaFetcherIntegrationTest.java

```diff
@@ -0,0 +1,421 @@
+          .setHeader("content-type", "application/vnd.uk.gov.api.v1alpha+json")
+          .setHeader("correlation-id", "response-correlation-id");
+    }
+
+    protected String baseUrl() {
+      return "%s/apis".formatted(server.url("").toString());
+    }
+  }
+
+  @Nested
+  class Fetch {
+    @Nested
+    class ___fTODO extends IntegrationTestBase { // TODO
+
+      @Test
+      void delegates__() {
+        var api =
+            new Api(
+                "The API Name",
+                "A longer description about the API",
+                "https://gov.uk",
+                "email@example.com",
+                "Department",
+                "https://docs");
+        var expected = new ApiMetadata();
+        expected.setApiVersion(ApiMetadata.ApiVersion.API_GOV_UK_V_1_ALPHA);
+        var data = new Data();
+        data.setName("T");
+        data.setName("The API Name");
+        data.setDescription("A longer description about the API");
+        data.setUrl(URI.create("https://gov.uk"));
+        data.setContact("email@example.com");
+        data.setOrganisation("Department");
+        data.setDocumentationUrl(URI.create("https://docs"));
+        expected.setData(data);
+
+        server.enqueue(success(api));
+
+        fetch();
+
+        verify(v1AlphaMapper, atLeast(1)).from(expected);
+      }
+
+      @Test
+      void returnsApisWhenSuccessful()
+          throws Fetcher.ClientErrorException, Fetcher.TemporaryErrorException,
+              Fetcher.VersionNotSupportedException {
+        var api =
+            new Api(
+                "The API Name",
+                "A longer description about the API",
+                "https://gov.uk",
+                "email@example.com",
+                "Department",
+                "https://docs");
+
+        server.enqueue(success(api));
+        when(v1AlphaMapper.from(any())).thenReturn(api);
+
+        var actual = fetch();
+
+        assertThat(actual).containsExactlyInAnyOrder(api);
+      }
+
+      @Test
+      void itAddsRequestCorrelationIdToMdcFromMdc() {
+        server.enqueue(success());
+        when(mdc.get(any())).thenReturn("request-correlation-id");
+
+        fetch();
+
+        verify(mdc, atLeast(1)).put("request.v1alpha.correlation-id", "request-correlation-id");
+      }
+
+      @Test
+      void itRetrievesKeyFromMdc() {
+        server.enqueue(success());
+
+        fetch();
+
+        verify(mdc, atLeast(1)).get("correlation-id");
+      }
+
+      @Test
+      void itAddsResponseCorrelationIdToMdc() {
+        server.enqueue(success());
+
+        fetch();
+
+        verify(mdc, atLeast(1)).put("response.v1alpha.correlation-id", "response-correlation-id");
+      }
+
+      @Test
+      void itLogsNumberOfItemsFetched() {
+        server.enqueue(success());
+
+        fetch();
+
+        assertThat(logger)
+            .anyThread()
+            .hasLogged(LoggingEvent.info("Successfully retrieved {} APIs from {}", 0, baseUrl()));
```

Comment:
do we want to add some valid options here to make sure it works?

Reply:
yeah sounds good